app/src/examples/RuntimeTests/ReanimatedRuntimeTestsRunner/TestRunner.ts

```diff
@@ -280,7 +310,39 @@ export class TestRunner {
       await testSuite.beforeEach();
     }
 
-    await testCase.run();
+    if (testCase.decorator === TestDecorator.FAILING || testCase.decorator === TestDecorator.WARN) {
+      const consoleTrackerRef = testCase.decorator === TestDecorator.FAILING ? 'console.error' : 'console.warn';
+      const message = makeMutable('');
+
+      const newConsoleFuncJS = (warning: string) => {
+        this.callTracker(consoleTrackerRef);
+        message.value = warning.split('\n\nThis error is located at:')[0];
+      };
+      console.error = newConsoleFuncJS;
+      console.warn = newConsoleFuncJS;
+
+      const callTrackerCopy = this.callTracker;
+
+      runOnUI(() => {
+        'worklet';
+        const newConsoleFuncUI = (warning: string) => {
+          callTrackerCopy(consoleTrackerRef);
+          message.value = warning.split('\n\nThis error is located at:')[0];
+        };
+        console.error = newConsoleFuncUI;
+        console.warn = newConsoleFuncUI;
+      })();
+
+      await testCase.run();
+
+      this.expect(getTrackerCallCount(consoleTrackerRef)).toBeCalled(1);
+      if (testCase.warningMessage) {
+        this.expect(message.value).toBe(testCase.warningMessage, ComparisonMode.STRING);
+      }
```

Comment:
The `expect` function should be utilized by the user during testing, not in our internal logic. This functionality is not the responsibility of the `expect` method. We can simply use a single shared value for it instead of a callTracker.